typed/remove_test.go

```diff
@@ -0,0 +1,133 @@
+
+	var cases = []struct {
+		object YAMLObject
+		remove *fieldpath.Set
+		deep   bool
+		expect YAMLObject
+	}{
+		{
+			object: `{}`,
+			remove: fieldpath.NewSet(fieldpath.MakePathOrDie("a")),
+			deep:   true,
+			expect: ``,
+		},
+		{
+			object: `{"a": "value"}`,
+			remove: fieldpath.NewSet(fieldpath.MakePathOrDie("a")),
+			deep:   true,
+			expect: `null`,
+		},
+		{
+			object: `{"a": "value", "b": "value"}`,
+			remove: fieldpath.NewSet(fieldpath.MakePathOrDie("a")),
+			deep:   true,
+			expect: `{"b": "value"}`,
+		},
+		{
+			object: `{"a": "value", "b": {}}`,
+			remove: fieldpath.NewSet(fieldpath.MakePathOrDie("a")),
+			deep:   true,
+			expect: `{"b": {}}`,
+		},
+		{
+			object: `{"a": "value", "b": {"c":"value"}}`,
+			remove: fieldpath.NewSet(fieldpath.MakePathOrDie("b")),
+			deep:   true,
+			expect: `{"a": "value"}`,
+		},
+		{
+			object: `{"a": "value", "b":{"c":"value"}}`,
+			remove: fieldpath.NewSet(fieldpath.MakePathOrDie("b")),
+			deep:   false,
+			// TODO: this seems to behave differently with a non deduced schema: {"a": "value", "b":{"c":"value"}} which is the reason for this PR
+			expect: `{"a": "value"}`,
+		},
+		{
+			object: `{"a": "value", "b": {"c":"value"}}`,
+			remove: fieldpath.NewSet(fieldpath.MakePathOrDie("b", "c")),
+			deep:   true,
+			// TODO: is a null result expected (insted of a {})
+			expect: `{"a": "value", "b": null}`,
+		},
+		{
+			object: `{"a": "value", "b": []}`,
+			remove: fieldpath.NewSet(fieldpath.MakePathOrDie("b")),
+			deep:   true,
+			expect: `{"a": "value"}`,
+		},
+		{
+			object: `{"a": "value", "b": ["c"]}`,
+			remove: fieldpath.NewSet(fieldpath.MakePathOrDie("b", "c")),
+			deep:   true,
+			expect: `{"a": "value", "b": null}`,
+		},
+		{
+			object: `{"a": "value", "b": ["c", "d"]}`,
+			remove: fieldpath.NewSet(fieldpath.MakePathOrDie("b", "c")),
+			deep:   true,
+			// TODO: is this expected?
+			expect: `{"a": "value", "b": null}`,
+		},
+		{
+			object: `{"a": "value", "b": [{"c": "value"}, {"d": "value"}]}`,
+			remove: fieldpath.NewSet(fieldpath.MakePathOrDie("b", "c")),
+			deep:   true,
+			// TODO: is this expected?
+			expect: `{"a": "value", "b": null}`,
+		},
+		{
+			object: `{"a": "value", "b": {"c":"value", "d":{"e":"value"}}}`,
+			remove: fieldpath.NewSet(fieldpath.MakePathOrDie("b", "d")),
+			deep:   true,
+			expect: `{"a": "value", "b": {"c":"value"}}`,
+		},
+		{
+			object: `{"a": "value", "b": {"c":"value", "d": {"e":"value"}}}`,
+			remove: fieldpath.NewSet(fieldpath.MakePathOrDie("b", "d", "e")),
+			deep:   true,
+			expect: `{"a": "value", "b": {"c":"value", "d": null}}`,
+		},
+	}
+
+	for i, c := range cases {
+		c := c
+		t.Run(fmt.Sprintf("case-%d", i), func(t *testing.T) {
+			t.Parallel()
+
+			obj, err := DeducedParseableType.FromYAML(c.object)
+			if err != nil {
+				t.Fatalf("unable to parse/validate object: %v\n%v", err, c.object)
+			}
+
+			expectTyped, err := DeducedParseableType.FromYAML(c.expect)
```

Comment:
These two `FromYAML` (here and below) look good to me. You might have to deep into the code a little bit more to undesrtand why you end-up with null instead of empty dir.